vcert/connection_tpp_abstract.py

```diff
@@ -561,8 +570,41 @@ def set_policy(self, zone, policy_spec):
         if len(prohibited_sans) > 0:
             self._set_policy_attr(name, SPA.TPP_PROHIBITED_SAN_TYPES, prohibited_sans, False)
 
+        # usernames = self.retrieve_usernames_from_tppcontacts(tpp_policy.name, tpp_policy)
+        # tpp_policy.contact(usernames)
+
         return
 
+    def resolve_tpp_contacts(self, contacts):
+        identities_id_list = list()
+        if contacts:
+            contacts = list(set(contacts))
+            for contact in contacts:
+                identity = self.get_tpp_identity(contact)
+                identities_id_list.append(identity.prefixed_universal)
+        return identities_id_list
+
+    def get_tpp_identity(self, username):
+        if not username or username == "":
+            raise VenafiError("Identity cannot be empty")
+        data = {
+            'Filter': username,
+            'Limit': 2,
+            'IdentityType': self.ALL_IDENTITIES
+        }
+        try:
+            status, response = self._post(URLS.POLICY_BROWSE_IDENTITY, data=data)
+            identities = response['Identities'][0]
+            if len(identities) == 0:
+                raise VenafiError(f"The username [{username}] ws not found.")
+        except VenafiError:
+            log.debug(f"Error while getting contact [{username}]")
+        if len(response['Identities']) > 1:
+            raise VenafiError("Extraneous information returned in the identity response. "
+                              "Expected size: 1, found: 2\n" + str(response['Identities'][1]))
```

Comment:
I think that the error message can't be that found 2 because you only knows that the size was major to 1